api/add.go

```diff
@@ -0,0 +1,115 @@
+package api
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/weaveworks/pctl/pkg/bootstrap"
+	"github.com/weaveworks/pctl/pkg/catalog"
+	"github.com/weaveworks/pctl/pkg/git"
+	"github.com/weaveworks/pctl/pkg/install"
+	"github.com/weaveworks/pctl/pkg/log"
+)
+
+// AddProfileOpts defines settings for adding profiles.
+type AddProfileOpts struct {
+	URL           string
+	Branch        string
+	SubName       string
+	Namespace     string
+	ConfigMap     string
+	Dir           string
+	Path          string
+	GitRepository string
+	ProfilePath   string
```

Comment:
This is the `catalog-name/profile-name` right? Might make sense to have them as separate strings now instead?

Reply:
If I take all that out into the CLI this looses most of its usefulness. Like, the point of this was that as the data comes in you call it with that and than this just takes care of the rest. That's the most useful part of being in here. If I remove all of those checks, there is nothing useful about this any longer. You could just construct `install` and call that like team wego already does here: https://github.com/weaveworks/weave-gitops-enterprise/pull/50

Reply:
https://github.com/weaveworks/weave-gitops-enterprise/pull/50/files#diff-24a52d493110426665082003b7a39ca6d6e2b5b75d1a162043f460c238c7a1fbR147

Reply:
I definitely see value though in adding validation for these values and the API taking care of that part.

Reply:
> If I take all that out into the CLI this looses most of its usefulness. Like, the point of this was that as the data comes in you call it with that and than this just takes care of the rest. 

but it sort of leaks the fact your using a CLI in the API right? If we imagine a world where this is happening in automation or through a UI this grouping makes less sense

> That's the most useful part of being in here. If I remove all of those checks, there is nothing useful about this any longer. 

I disagree. The value is just having the API, and validating the args.

Reply:
the fact they are together is a technical implementation of our CLI though. In WeGo maybes its `--git-repository-name` and `--git-repository-namespace`, and then they have to concatenate it into our format to use our API?

Reply:
Okay, yeah, that's a fair point. :)